framework/src/main/java/org/checkerframework/common/aliasing/AliasingVisitor.java

```diff
@@ -271,7 +265,37 @@ private boolean canBeLeaked(Tree exp) {
         AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(exp);
         boolean isMethodInvocation = exp.getKind() == Kind.METHOD_INVOCATION;
         boolean isNewClass = exp.getKind() == Kind.NEW_CLASS;
-        return type.hasExplicitAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        boolean isUniqueClassFlag = isUniqueClass(exp);
+        if (isUniqueClassFlag) {
```

Comment:
Could this just be `return !isMethodInvocation && !isNewClass && (isUniqueClass || type.hasExplicitAnnotation(Unique.class));`

Reply:
Maybe extract the `(isUniqueClass || type.hasExplicitAnnotation(Unique.class)` into a separate local and document why that is the logic.

Reply:
I have separated the (isUniqueClass || type.hasExplicitAnnotation(Unique.class) into a separate local variable and documented the logic.